ci/build.py

```diff
@@ -0,0 +1,106 @@
+import os
+from subprocess import call
+from pathlib import Path
+import re
+
+import yaml
+
```

Comment:
I refactored this file, splitting it into smaller functions, so that I can have more fined grained tests reports on CI, and so that I can run CI on some files only.